plugin/ochttp/trace.go

```diff
@@ -53,11 +56,80 @@ func (t *traceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	name := spanNameFromURL(req.URL)
 	// TODO(jbd): Discuss whether we want to prefix
 	// outgoing requests with Sent.
-	_, span := trace.StartSpan(req.Context(), name,
+	ctx, span := trace.StartSpan(req.Context(), name,
 		trace.WithSampler(t.startOptions.Sampler),
 		trace.WithSpanKind(trace.SpanKindClient))
 
-	req = req.WithContext(trace.WithSpan(req.Context(), span))
+	tracer := &httptrace.ClientTrace{
```

Comment:
Does this need to be allocated afresh each time or can it be statically allocated e.g. in a package global?

Reply:
We need to freshly allocate in order to reuse the context passed in between intermediate trace steps.